stetho/src/main/java/com/facebook/stetho/inspector/elements/android/AndroidDocumentProvider.java

```diff
@@ -89,8 +93,13 @@ public AndroidDocumentProvider(Application application, ThreadBound enforcer) {
         .register(TextView.class, new TextViewDescriptor())
         .register(View.class, new ViewDescriptor())
         .register(ViewGroup.class, new ViewGroupDescriptor())
-        .register(Window.class, new WindowDescriptor())
-        .setHost(this)
+        .register(Window.class, new WindowDescriptor());
+
+    for (DescriptorProvider descriptorProvider : descriptorProviders) {
```

Comment:
Since we have a `List`, we should avoid the temporary `Iterator` allocation by looping like so:

```
for (int i = 0, size = descriptorProviders.size(); i < size; ++i) {
  final DescriptorProvider descriptorProvider = descriptorProviders.get(i);
  ...
}
```